src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -160,16 +162,27 @@ class VisitService(
     //  Add dummy visitor row for the offender_booking as is required by the P-Nomis view
 
     visit.visitors.add(
-      VisitVisitor(visit = visit, offenderBooking = offenderBooking, eventStatus = scheduledEventStatus)
+      VisitVisitor(
+        visit = visit,
+        offenderBooking = offenderBooking,
+        eventStatus = scheduledEventStatus,
+        eventId = getNextEvent()
+      )
     )
 
     visitDto.visitorPersonIds.forEach {
       val person = personRepository.findById(it).orElseThrow(DataNotFoundException("Person with id=$it does not exist"))
       visit.visitors.add(
-        VisitVisitor(visit = visit, person = person, eventStatus = scheduledEventStatus)
+        VisitVisitor(visit = visit, person = person, eventStatus = scheduledEventStatus, eventId = getNextEvent())
       )
```

Comment:
do you think you need it on each visitor as well as the dummy row?

Reply:
Ah good point - no, I think actually only needed on dummy row , will refine that next time